67_RayQueryGeometry/main.cpp

```diff
@@ -590,6 +597,235 @@ class RayQueryGeometryApp final : public examples::SimpleWindowedApplication, pu
+					inst.base.mask = 0xFF;
+
+					core::matrix3x4SIMD transform;
+					transform.setTranslation(nbl::core::vectorSIMDf(5.f * i, 0, 0, 0));
+					inst.transform = transform;
+					
+					instance->instance = inst;
+				}
+			}
+
+			smart_refctd_ptr<ICPUTopLevelAccelerationStructure> cpuTlas;
+			cpuTlas->setInstances(std::move(geomInstances));
+			cpuTlas->setBuildFlags(IGPUTopLevelAccelerationStructure::BUILD_FLAGS::PREFER_FAST_TRACE_BIT);
+
+			// convert with asset converter
+			auto pool = m_device->createCommandPool(queue->getFamilyIndex(), IGPUCommandPool::CREATE_FLAGS::RESET_COMMAND_BUFFER_BIT);
+			if (!pool)
+				return logFail("Couldn't create Command Pool for geometry creation!");
+			auto cmdbuf = getSingleUseCommandBufferAndBegin(pool);
+			cmdbuf->beginDebugMarker("Build geometry vertex and index buffers");
+
+			smart_refctd_ptr<CAssetConverter> converter = CAssetConverter::create({ .device = m_device.get(), .optimizer = {} });
+			CAssetConverter::SInputs inputs = {};
+			inputs.logger = m_logger.get();
+
+			std::array<ICPUTopLevelAccelerationStructure*, 1u> tmpTlas;
+			std::array<ICPUBuffer*, OT_COUNT * 2u> tmpBuffers;
+			{
+				tmpTlas[0] = cpuTlas.get();
+				for (uint32_t i = 0; i < objectsCpu.size(); i++)
+				{
+					tmpBuffers[2 * i + 0] = scratchBuffers[i].vertex.buffer.get();
+					tmpBuffers[2 * i + 1] = scratchBuffers[i].index.buffer.get();
+				}
+
+				std::get<CAssetConverter::SInputs::asset_span_t<ICPUTopLevelAccelerationStructure>>(inputs.assets) = tmpTlas;
+				std::get<CAssetConverter::SInputs::asset_span_t<ICPUBuffer>>(inputs.assets) = tmpBuffers;
+			}
+
+			auto reservation = converter->reserve(inputs);
+			{
+				auto prepass = [&]<typename asset_type_t>(const auto & references) -> bool
+				{
+					auto objects = reservation.getGPUObjects<asset_type_t>();
+					uint32_t counter = {};
+					for (auto& object : objects)
+					{
+						auto gpu = object.value;
+						auto* reference = references[counter];
+
+						if (reference)
+						{
+							if (!gpu)
+							{
+								m_logger->log("Failed to convert a CPU object to GPU!", ILogger::ELL_ERROR);
+								return false;
+							}
+						}
+						counter++;
+					}
+					return true;
+				};
+
+				prepass.template operator() < ICPUTopLevelAccelerationStructure > (tmpTlas);
+				prepass.template operator() < ICPUBuffer > (tmpBuffers);
+			}
+
+			auto semaphore = m_device->createSemaphore(0u);
+
+			std::array<IQueue::SSubmitInfo::SCommandBufferInfo, 1> cmdbufs = {};
+			cmdbufs.front().cmdbuf = cmdbuf.get();
+
+			SIntendedSubmitInfo transfer = {};
+			transfer.queue = queue;
+			transfer.scratchCommandBuffers = cmdbufs;
+			transfer.scratchSemaphore = {
+				.semaphore = semaphore.get(),
+				.value = 0u,
+				.stageMask = PIPELINE_STAGE_FLAGS::ALL_TRANSFER_BITS	// TODO mask for AS?
+			};
+			// convert
+			{
+				CAssetConverter::SConvertParams params = {};
+				params.utilities = m_utils.get();
+				params.transfer = &transfer;
```

Comment:
you need two different intended submit infos (referencing a different ring of vcommandbuffers), one for transfer and one for compute, even if you're using the same queue for everything